services/QuillLMS/client/app/bundles/Staff/components/comprehension/activitySessions/sessionsIndex.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from "react";
+import { Link } from 'react-router-dom';
+import { useQuery } from 'react-query';
+import * as moment from 'moment'
+
+import { DataTable, Error, Spinner, DropdownInput } from '../../../../Shared/index';
+import { fetchActivity, fetchActivitySessions } from '../../../utils/comprehension/activityAPIs';
+import { DropdownObjectInterface } from '../../../interfaces/comprehensionInterfaces';
+
+const quillCheckmark = 'https://assets.quill.org/images/icons/check-circle-small.svg';
+
+const SessionsIndex = ({ match }) => {
+  const { params } = match;
+  const { activityId } = params;
+
+  const [pageNumber, setPageNumber] = React.useState<DropdownObjectInterface>(null);
+  const [dropdownOptions, setDropdownOptions] = React.useState<DropdownObjectInterface[]>(null);
+  const pageNumberForQuery = pageNumber && pageNumber.value ? pageNumber.value : 1;
+
+  // cache activity data for updates
+  const { data: activityData } = useQuery({
+    queryKey: [`activity-${activityId}`, activityId],
+    queryFn: fetchActivity
+  });
+
+  // cache activity sessions data for updates
+  const { data: sessionsData } = useQuery({
+    queryKey: [`activity-${activityId}-sessions`, activityId, pageNumberForQuery],
```

Comment:
Doesn't the `pageNumberForQuery` also need to be part of the queryKey string if we want it to update when it changes?

Reply:
no, this query-string will be unique to the sessions data of this activity as is. If we wanted to manually refetch the query for whatever reason, we could do so like this:
```
queryClient.refetchQueries([`activity-${activityId}-sessions`, activityId,  pageNumberForQuery]) 
```